core/trino-main/src/main/java/io/trino/operator/scalar/SessionFunctions.java

```diff
@@ -64,4 +69,17 @@ public static Slice currentSchema(ConnectorSession session)
                 .map(Slices::utf8Slice)
                 .orElse(null);
     }
+
+    @ScalarFunction("current_groups")
+    @Description("Current groups of current user")
+    @SqlType("array(varchar)")
+    public static Block currentGroups(ConnectorSession session)
```

Comment:
@martint any reason why this implementation is different from `current_user` (https://github.com/trinodb/trino/pull/7829/files)

Is it because this is not part of SQL standard so can be just a function instead of a literal `SELECT current_group()` vs `SELECT current_groups`?

Reply:
`CURRENT_USER` is a language feature defined by the SQL specification. There's not such a thing for `current_groups`, so we have to implement it as a regular function.